core/primitives/src/sharding.rs

```diff
@@ -439,12 +460,23 @@ impl ShardChunkHeader {
         }
     }
 
+    #[inline]
+    pub fn permanent_contracts_metadata(&self) -> &[(AccountId, CryptoHash)] {
```

Comment:
to be consistent in naming, can be rename `permanent_contracts_metadata` to `global_contracts_metadata`?